src/document.c

```diff
@@ -1955,37 +1961,51 @@ static gchar *write_data_to_disk(const gchar *locale_filename,
 		/* Use POSIX API for unsafe saving (GVFS-unsafe) */
 		/* The error handling is taken from glib-2.26.0 gfileutils.c */
 		errno = 0;
-		fp = g_fopen(locale_filename, "wb");
+		fp = g_fopen(locale_filename, "w+b"); // we can not destroy  symbolic/hard links
 		if (fp == NULL)
 		{
 			save_errno = errno;
 
 			g_set_error(&error,
 				G_FILE_ERROR,
 				g_file_error_from_errno(save_errno),
-				_("Failed to open file '%s' for writing: fopen() failed: %s"),
+				_("Failed to open file '%s' for update-content: fopen() failed: %s"),
 				display_name,
 				g_strerror(save_errno));
 		}
 		else
 		{
-			gsize bytes_written;
-
-			errno = 0;
-			bytes_written = fwrite(data, sizeof(gchar), len, fp);
-
-			if (len != bytes_written)
+			if(truncate(locale_filename,sizeof(gchar)*len)) // no can change file size error!
```

Comment:
Glib makes locale_filename UTF8 on windows, you can't pass it to a clib function I don't think.